### PR TITLE
Change time format in suffix of run directory

### PR DESCRIPTION
**Pull request by portante**
Currently the time format used in the directory names for the benchmark
run directory is:

    TS_FORMAT="%FT%H:%M:%S"

We'd like to change it to:

    TS_FORMAT="%Y.%m.%dT%H.%M.%S"

Making this change allows us to pass the benchmark run directory to
docker to bind mount into a container.

Currently, when trying to export $benchmark_run_dir inside container,
E.g.:

    docker run --privileged -it -v $benchmark_run_dir:/pbench image bash

it will fail with:

    /usr/bin/docker-current: Error response from daemon: Invalid bind
    mount spec "/var/lib/pbench-agent/pbench-user-benchmark_sysbench"
    "incontainer_2017-04-11_15:17:06:/pbench": Invalid volume specif"
    "ication: '/var...

The colon in the directory name is not handled properly by docker as,
that is interpreted by docker as point to look for source:destination
when "-v" volume flag is used.

Signed-off-by: Elvir Kuric <elvirkuric@gmail.com>

### CODE REVIEW

**Comment by portante** [comment]
Supercedes #550.

**Comment by portante** [comment]
@ekuric, I rewrote your original pull request to include unit test updates, and to address the problem slightly differently, so that we don't loose the existing timestamp format in the metadata log file.

File: agent/base [hunk 1]
```diff
@@ -90,13 +90,16 @@ if [[ -z "$_PBENCH_BENCH_TESTS" ]]; then
     # to check whether we need to set it.
     if [ -z "$date" -o -z "$_PBENCH_DATE_SET" ] ;then
         # don't use ns in the date
-        export date=`date --utc +"${TS_FORMAT}"`
+        export date=$(date --utc +"%Y-%m-%dT%H:%M:%S")
```
Comment (ndokos) [inline agent/base]: Shouldn't this be `${TS_FORMAT}`?

File: agent/bench-scripts/pbench-netperf [hunk 2]
```diff
@@ -337,6 +336,20 @@ while true; do
...
 
+if [[ -z "$benchmark_run_dir" ]]; then
+        # We don't have an explicit run directory, construct one
+        benchmark_fullname="${benchmark}_${config}_${date_suffix}"
+        benchmark_run_dir="$pbench_run/${benchmark_fullname}"
+else
+        # We have an explicit run directory provided by --run-dir, so warn
+        # the user if they also used --config
+        if [[ ! -z "$config" ]]; then
+                warn_log "[$script_name] ignoring --config=\"$config\" in favor of --rundir=\"$benchmark_run_dir\""
```
Comment (ndokos) [inline agent/bench-scripts/pbench-netperf]: --run-dir

**Pull request merged**